pkg/node/node.go

```diff
@@ -366,36 +363,26 @@ func NewBee(o Options) (*Bee, error) {
 		b.debugAPIServer = debugAPIServer
 	}
 
-	overlays, err := addressbook.Overlays()
+	addresses, err := addressbook.Addresses()
 	if err != nil {
 		return nil, fmt.Errorf("addressbook overlays: %w", err)
 	}
 
 	var count int32
-	var wg sync.WaitGroup
-	jobsC := make(chan struct{}, 16)
-	for _, o := range overlays {
-		jobsC <- struct{}{}
-		wg.Add(1)
-		go func(overlay swarm.Address) {
-			defer func() {
-				<-jobsC
-			}()
-
-			defer wg.Done()
-			if err := topologyDriver.AddPeer(p2pCtx, overlay); err != nil {
-				logger.Debugf("topology add peer fail %s: %v", overlay, err)
-				logger.Warningf("topology add peer %s", overlay)
-				return
-			}
 
-			atomic.AddInt32(&count, 1)
-		}(o)
+	// add the peers to topology and allow it to connect independently
+	for _, o := range addresses {
+		err = topologyDriver.AddPeer(p2pCtx, o.Overlay)
+		if err != nil {
+			logger.Debugf("topology add peer from addressbook: %v", err)
+		} else {
+			count++
+		}
```

Comment:
it's true that we can adjust `AddPeer` to be a variadic function and it can return a boolean or error array, but this is an optimization

Reply:
i guess it is indeed an optimization, but i would argue it would make the code tidier and more maintainable.

it is not a blocker for me anyways, so i'll resolve

Reply:
we need `count` for the bootnodes below... although, as said, this should probably be handled somehow through kademlia...